libs/text-splitters/langchain_text_splitters/html.py

```diff
@@ -6,7 +6,9 @@
 from typing import Any, Dict, Iterable, List, Optional, Tuple, TypedDict, cast
 
 import requests
-from langchain_core.documents import Document
+from bs4 import BeautifulSoup
```

Comment:
Move these imports into local scope. Most of langchain implementations have to avoid assuming that a particular 3rd party package is installed on a user's machine

```python
from bs4 import BeautifulSoup
from bs4.element import Tag
```
This import is incorrect, use the `langchain_core.documents import Document` import that you had before

```python
from langchain.docstore.document import Document
```